api/_db.data-utils.js

```diff
@@ -55,6 +55,27 @@ const parseDate = (rawValue) => {
 	return parsed;
 };
 
+const findHighestValidCertificate = (certificates) => {
+	// find the certificate that is not expired and has the highest value (red is higher than green).
+	// nonexpired green has to beat expired red, but red with shorter expiryTime has to beat any green.
+	return certificates.reduce((previous, current) =>{
+		if (!previous) {
+			return current;
+		}
+		if (parseDate(current.expiryTime) > Date.now()){
+			// TODO: maybe we should find the certificate with the longest expiry time
+			// currently the first valid one is kept
+			if (previous.certificate != CODE.RED && current.certificate == CODE.RED){
+				return current;
+			}
+			if (previous.certificate == CODE.GREEN && current.certificate == CODE.GREEN){
```

Comment:
This makes an assumption that order of which the data is presented(not "entered" because the process currently does not really assert order) to this function matters. Do we really want to make that assumption?